pybamm/models/submodels/particle/polynomial_single_particle.py

```diff
@@ -0,0 +1,356 @@
+            # than DAEs.
+            if self.domain == "Negative":
+                j_xav = i_boundary_cc / self.param.l_n
+                c_s_surf_xav = c_s_rxav - self.param.C_n * (
+                    j_xav
+                    / 5
+                    / self.param.a_n
+                    / self.param.D_n(c_s_rxav, pybamm.surf(T_xav))
+                )
+
+            if self.domain == "Positive":
+                j_xav = -i_boundary_cc / self.param.l_p
+                c_s_surf_xav = c_s_rxav - self.param.C_p * (
+                    j_xav
+                    / 5
+                    / self.param.a_p
+                    / self.param.gamma_p
+                    / self.param.D_p(c_s_rxav, pybamm.surf(T_xav))
+                )
+        elif self.order == 4:
+            # The surface concentration is computed from the average concentration,
+            # the average concentration gradient and the boundary flux (see notes
+            # for the case order=2)
+            q_s_rxav = variables[
+                "R-X-averaged "
+                + self.domain.lower()
+                + " particle concentration gradient"
+            ]
+            if self.domain == "Negative":
+                j_xav = i_boundary_cc / self.param.l_n
+                c_s_surf_xav = (
+                    c_s_rxav
+                    + 8 * q_s_rxav / 35
+                    - self.param.C_n
+                    * (
+                        j_xav
+                        / 35
+                        / self.param.a_n
+                        / self.param.D_n(c_s_rxav, pybamm.surf(T_xav))
+                    )
+                )
+
+            if self.domain == "Positive":
+                j_xav = -i_boundary_cc / self.param.l_p
+                c_s_surf_xav = (
+                    c_s_rxav
+                    + 8 * q_s_rxav / 35
+                    - self.param.C_p
+                    * (
+                        j_xav
+                        / 35
+                        / self.param.a_p
+                        / self.param.gamma_p
+                        / self.param.D_p(c_s_rxav, pybamm.surf(T_xav))
+                    )
+                )
+
+        # Set concentration depending on polynomial order
+        if self.order == 0:
+            # The concentration is uniform
+            c_s_xav = pybamm.PrimaryBroadcast(
+                c_s_rxav, [self.domain.lower() + " particle"]
+            )
+        elif self.order == 2:
+            # The concentration is given by c = A + B*r**2
+            A = pybamm.PrimaryBroadcast(
+                (1 / 2) * (5 * c_s_rxav - 3 * c_s_surf_xav),
+                [self.domain.lower() + " particle"],
+            )
+            B = pybamm.PrimaryBroadcast(
+                (5 / 2) * (c_s_surf_xav - c_s_rxav), [self.domain.lower() + " particle"]
+            )
+            if self.domain == "Negative":
+                # TODO: figure out how to just use r from standard_spatial_vars
```

Comment:
I kept getting shape errors here and couldn't figure out the correct way to broadcast things so I could use `r` from `standard_spatial_vars` here. I thought I should broadcast A, B, (C), then use that to define `c_s`, and then average to get `c_s_av`, but I kept getting bogged down in shape errors

Reply:
Since `c_s_xav` doesn't depend on `x`, it makes sense that you should use a different `r` which only has "negative particle" and "current collector" as domains